utils/rpc/json.go

```diff
@@ -49,15 +50,27 @@ func SendJSONRequest(
 
 	// Return an error for any non successful status code
 	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		// Avoid sending unnecessary RST_STREAM and PING frames by ensuring the whole body is read.
+		// See https://blog.cloudflare.com/go-and-enhance-your-calm/#reading-bodies-in-go-can-be-unintuitive
+		_, _ = io.Copy(io.Discard, resp.Body)
```

Comment:
If the server is malicious they can dump an infinite length response body into us. We're not in danger of OOM'ing because `io.Copy` allocates a fixed buffer, but we could spin indefinitely. I think a more robust strategy might be to drain the response body if we can, or send a `RST_STREAM` to the server if the caller cancels our context.